Content.Server/Health/BodySystem/Surgery/SurgeryPerformer/SurgeryToolComponent.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System;
+using System.Collections.Generic;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Shared.BodySystem;
+using Content.Shared.GameObjects;
+using Content.Shared.Interfaces;
+using Robust.Server.GameObjects;
+using Robust.Server.GameObjects.Components.UserInterface;
+using Robust.Server.Interfaces.Player;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Serialization;
+
+namespace Content.Server.BodySystem
+{
+    //TODO: add checks to close UI if user walks too far away from tool or target.
+
+    /// <summary>
+    ///     Server-side component representing a generic tool capable of performing surgery. For instance, the scalpel.
+    /// </summary>
+    [RegisterComponent]
+    public class SurgeryToolComponent : Component, ISurgeon, IAfterInteract
+    {
+        public override string Name => "SurgeryTool";
+        public override uint? NetID => ContentNetIDs.SURGERY;
+
+#pragma warning disable 649
+        [Dependency] private readonly ISharedNotifyManager _sharedNotifyManager;
+        [Dependency] private readonly IRobustRandom _random;
+#pragma warning restore 649
+
+        public float BaseOperationTime { get => _baseOperateTime; set => _baseOperateTime = value; }
+        private float _baseOperateTime;
+        private SurgeryType _surgeryType;
+        private HashSet<IPlayerSession> _subscribedSessions = new HashSet<IPlayerSession>();
+        private BoundUserInterface _userInterface;
+
+        private Dictionary<object, object> _optionsCache = new Dictionary<object, object>();
```

Comment:
Similar question for this - looks like `<int, Mechanism>`?

Reply:
Fixed